Src/VimCore/VimTextBuffer.fs

```diff
@@ -23,6 +23,8 @@ type internal VimTextBuffer
     let _switchedModeEvent = StandardEvent<SwitchModeKindEventArgs>()
     let mutable _modeKind = ModeKind.Normal
     let mutable _lastVisualSelection : ITrackingVisualSelection option = None
+    let mutable _lastModifiedSpan : ITrackingVisualSelection option = None
+    let mutable _lastChangedOrYankedSpan : SnapshotSpan option = None

```

Comment:
The type of this value should be `ITrackingSpan`.  

**Long explanation for those unfamiliar with the editor model here:**

The `SnapshotSpan` type represents a span of text with in a specific snapshot (represented as `ITextSnapshot`) of the buffer.  A snapshot is immutable and can be safely held onto for long periods of time.  Every edit to the buffer (represented as `ITextBuffer`) produces a new snapshot value.  

Operations such as caret move, edits, etc ...  are all done in terms of the current snapshot of the buffer.  If you attempt to say move the caret using a `SnapshotPoint` from a previous snapshot it will throw an exception.  

Storing the last modified span as a `SnapshotSpan` value means it will become invalid on the very next edit.  Any attempt to use <kbd>`[</kbd> at that point will result in an exception.

Types like `ITrackingSpan` are meant for storing spans which need to adjust as the buffer is edited.  The `TrackingSpanUtil` type makes transitions between `SnapshotSpan` and `ITrackingSpan` pretty easy.

``` fsharp
let trackingSpan = TrackingSpanUtil.Create snapshotSpan SpanTrackingMode.EdgeInclusive
```